src/main/java/com/bandwidth/sqs/publisher/MessagePublisher.java

```diff
@@ -0,0 +1,12 @@
+package com.bandwidth.sqs.publisher;
+
+import com.amazonaws.services.sqs.model.SendMessageResult;
+
+import java.time.Duration;
+import java.util.Optional;
+
+import io.reactivex.Single;
+
+public interface MessagePublisher<T> {
+    Single<SendMessageResult> publishMessage(T newMessage, String newQueueUrl, Optional<Duration> delay);
```

Comment:
0 is different than 'empty'. Empty mean to use the default of the queue, where 0 is 0.
I documented this, but I guess I should move that documentation to the interface instead of one of the implementations...